torchrl/trainers/helpers/envs.py

```diff
@@ -428,6 +446,78 @@ def get_stats_random_rollout(
     return stats
 
 
+def initialize_observation_norm_transforms(
+    proof_environment: EnvBase,
+    num_iter: int = 1000,
+    key: Union[str, Tuple[str, ...]] = None,
+):
+    """Calling init_stats on all uninitialised ObservationNorms transform of a TransformedEnv.
+
+    If an ObservationNorm already has non-null loc or stats, it will be skipped.
+    If the transform of the environment does not contain any ObservationNorm, nothing will happen.
+    If no key is provided but the observations of the Environment contains more than one key, an exception will
+    be raised.
+
+    Args:
+        proof_environment (EnvBase instance, optional): if provided, this env will
+            be used ot execute the rollouts. If not, it will be created using
+            the cfg object.
+        num_iter (int): Number of iterations used for initializing the ObservationNorms
+        key (str, optional): if provided, the stats of this key will be gathered.
+            If not, it is expected that only one key exists in `env.observation_spec`.
+
+    """
+    if not isinstance(proof_environment.transform, Compose) and not isinstance(
+        proof_environment.transform, ObservationNorm
+    ):
+        return
+
+    if key is None:
+        keys = list(proof_environment.observation_spec.keys())
+        key = keys.pop()
+        if len(keys):
+            raise RuntimeError(
+                f"More than one key exists in the observation_specs: {[key] + keys} were found, "
+                "thus generate_stats_from_observation_norm cannot infer which to compute the stats of."
+            )
+
+    if isinstance(proof_environment.transform, Compose):
+        for transform in proof_environment.transform:
+            if (
+                isinstance(transform, ObservationNorm)
+                and transform.loc is None
+                and transform.scale is None
+            ):
+                transform.init_stats(num_iter=num_iter, key=key)
+    elif (
+        proof_environment.transform.loc is None
+        and proof_environment.transform.scale is None
+    ):
+        proof_environment.transform.init_stats(num_iter=num_iter, key=key)
+
+
+def retrieve_observation_norms_state_dict(proof_environment: TransformedEnv):
+    """Traverse the transforms of the environment and retrieve the ObservationNorm state dicts.
```

Comment:
Maybe Traverses + retrieves?

Also for code objects we can use :obj:`ObservationNorm` for a cleaner rendering in the doc